client/src/components/tools/SortDropDownMenu.tsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom"
+
+type Props = {
+    setParams: Function,
+    options: string[],
+    title: string
+  };
+
+const SortDropDownMenu = ({ setParams, options, title }: Props) => {
+
+  return(<>
+    {title}
+    <select
+      className="ml-5 mt-2 border"
+      onChange={(event) => setParams(event.target.value)}
+    >
+        <option className="bg-red-500" value="" selected disabled hidden>
+          Select
+        </option>
+        {options.map((option) => (
+          <option key={option} value={`${option}`}>{option}</option>
```

Comment:
![image](https://user-images.githubusercontent.com/83677410/229031275-e989f34e-c6f4-450d-8bf1-ce9b1b4dfd48.png)

Can the displayed values be formatted? Some quick format functions in should do the trick, ( you could export them from some `stringUtilities` file in `/shared`... )

```ts
function fromSnakeCase(value: string) {
  return value.replace(/_/g, " ");
}

function toSentenceCase(value: string) {
  return value.charAt(0).toUpperCase() + value.slice(1);
}
```

```tsx
{options.map((option) => {
  const displayOption = fromSnakeCase(toSentenceCase(option));
  return (
    <option key={option} value={option}>
      {displayOption}
    </option>
  );
})}
```

![image](https://user-images.githubusercontent.com/83677410/229032963-86c00c41-1dd9-4f56-8fa6-b9103cf5c9ee.png)

Reply:
Nice idea, I have implemented this.  Typescript needed a later language version to enable the replaceAll function, I changed it to es2021, I'm not sure why it wouldn't be a later version by default, are there any obvious issues doing this?

Reply:
Strange. ES2021 should be fine.